src/term.rs

```diff
@@ -858,15 +863,24 @@ impl RichTerm {
         self
     }
 
-    /// Apply a transformation on a whole term by mapping a function `f` on each node in a
-    /// bottom-up manner. `f` may return a generic error `E` and use the state `S` which is
-    /// passed around.
-    pub fn traverse<F, S, E>(self, f: &mut F, state: &mut S) -> Result<RichTerm, E>
+    /// Apply a transformation on a whole term by mapping a function `f` on each node in
+    /// manner defined by the method.
+    /// `f` may return a generic error `E` and use the state `S` which is passed around.
+    pub fn traverse<F, S, E>(
+        self,
+        f: &mut F,
+        state: &mut S,
+        method: &TraverseMethod,
```

Comment:
We should probably make `TraverseMethod` derive `Copy` and never care anymore about references and borrows.